library/ecdsa.c

```diff
@@ -635,7 +635,7 @@ static int ecdsa_verify_restartable( mbedtls_ecp_group *grp,
     if( mbedtls_mpi_cmp_int( r, 1 ) < 0 || mbedtls_mpi_cmp_mpi( r, &grp->N ) >= 0 ||
         mbedtls_mpi_cmp_int( s, 1 ) < 0 || mbedtls_mpi_cmp_mpi( s, &grp->N ) >= 0 )
     {
-        ret = MBEDTLS_ERR_ECP_VERIFY_FAILED;
+        ret = MBEDTLS_ERR_ECP_BAD_INPUT_DATA;
```

Comment:
Either error code is arguably correct. In an LTS, we should err on the side of being conservative.